src/bin/add/main.rs

```diff
@@ -70,20 +72,46 @@ crates.io registry suggests. One goal of `cargo add` is to prevent you from usin
 dependencies (version set to "*").
 "#;
 
+fn print_msg(dep: &Dependency, section: &[String], optional: bool) {
+    let optional = if optional { "optional " } else { "" };
+    let section = if section.len() == 1 {
+        section[0].clone()
+    } else {
+        format!("{} for target {}", &section[2], &section[1])
+    };
+
+    let mut output = StandardStream::stdout(ColorChoice::Auto);
+    output
+        .set_color(ColorSpec::new().set_fg(Some(Color::Green)).set_bold(true))
+        .unwrap();
+    print!("{:>12}", "Adding");
+    output.reset().unwrap();
+    println!(
+        " {} v{} to {}{}",
+        dep.name,
+        dep.version().unwrap_or(""),
```

Comment:
This will look a bit odd when `dep.version()` is `None`. I think you will get e.g. `Adding foo v to dependencies`. Should we avoid printing the `v` when we don't know the version?

Reply:
Indeed. Do you know in which cases the version is ever `None`? Do we retrieve version from git dependencies?

I suggest splitting this one `println!` into several `print!`s:

```rust
print!(" {}", name);
if let Some(version) = dep.version() {
    print!(" v{}", version)
} else {
    print!(" (unknown version)"); // maybe write something useful here instead
}
print!(" to")
if optional {
    print!(" optional");
}
print!(" {}", section);
println!(); // just for the newline
```

Reply:
`version()` returns `None` iff we have either a git dependency, or a path dependency.
We can implement `Display` for `Dependency`, which will render dependency to the appropriate string `v0.1.0`/`(git://foo.bar.git)`/`../path` as mentioned in the comment [above](https://github.com/killercup/cargo-edit/pull/190#issuecomment-355437598).

Reply:
Good idea. But let's do that in a separate PR and land this with no version when `version()` is `None`.